src/index.ts

```diff
@@ -96,21 +96,29 @@ if ( config.googleSSOEnabled ) {
     async (req, issuer, profile, done) => {
       const googleEmail = profile.emails[0].value.trim();  // Get Google email
 
-      try {
-        const doc = await db.get(req.session.passport.user)
-        if (doc.oauthConnections) {
-          doc.oauthConnections.google = googleEmail
-        }
-        else {
-          doc.oauthConnections = {}
-          doc.oauthConnections.google = googleEmail
+      let docs = await db.find({
+          selector: { "oauthConnections.google": { $eq: googleEmail } }
+      });
+      if (docs.docs.length == 1 ){
+        req.flash('error', _CC.lang('LOGIN_SSO_LINK_FAILURE_ACCOUNT_EXISTS'))
+        return done(null)
+      } else {
+        try {
+          const doc = await db.get(req.session.passport.user)
+          if (doc.oauthConnections) {
```

Comment:
My preference is for this to be done like this:
```js
doc.oauthConnections ??= {} // handle migration from 1.35 and earlier releases
doc.oauthConnections.google = googleId // TODO: call the variable googleId rather than googleEmail
```

There's also an argument to be made for the following, since it doesn't use newish/obscure JS features. I prefer the above though.
```js
if (typeof doc.oauthConnections === 'undefined') doc.oauthConnections = {} // handle migration from 1.35 and earlier releases
doc.oauthConnections.google = googleId
```

Reply:
Agree. Updated with https://github.com/Wingysam/Christmas-Community/pull/138/commits/da1375887114c48ea581fb5a91d46f3891d563b4